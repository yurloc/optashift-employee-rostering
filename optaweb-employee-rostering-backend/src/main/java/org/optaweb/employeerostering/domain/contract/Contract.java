/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.optaweb.employeerostering.domain.contract;

import javax.persistence.Entity;
import javax.persistence.Table;
import javax.persistence.UniqueConstraint;
import javax.validation.constraints.AssertTrue;
import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.optaweb.employeerostering.domain.common.AbstractPersistable;

@Entity
@Table(uniqueConstraints = {@UniqueConstraint(columnNames = {"tenantId", "name"}),
        @UniqueConstraint(columnNames = {"id"})})
// TODO: Single Responsibility Principle - acts as both domain entity and JSON-serializable entity
public class Contract extends AbstractPersistable {

    @NotNull
    @Size(min = 1, max = 120)
    @Pattern(regexp = "^(?!\\s).*(?<!\\s)$", message = "Name should not contain any leading or trailing whitespaces")
    private String name;

    // Can be null
    @Min(0)
    @Max(60 * 24)
    private Integer maximumMinutesPerDay;

    // Can be null
    @Min(0)
    @Max(60 * 24 * 7)
    private Integer maximumMinutesPerWeek;

    // Can be null
    // 31 days in longest month
    @Min(0)
    @Max(60 * 24 * 31)
    private Integer maximumMinutesPerMonth;

    // Can be null
    // 366 days in leap year
    @Min(0)
    @Max(60 * 24 * 366)
    private Integer maximumMinutesPerYear;

    @SuppressWarnings("unused")
    public Contract() {
    }

    public Contract(Integer tenantId, String name) {
        this(tenantId, name, null, null, null, null);
    }

    public Contract(Integer tenantId, String name, Integer maximumMinutesPerDay, Integer maximumMinutesPerWeek,
                    Integer maximumMinutesPerMonth, Integer maximumMinutesPerYear) {

        super(tenantId);
        this.name = name;
        this.maximumMinutesPerDay = maximumMinutesPerDay;
        this.maximumMinutesPerWeek = maximumMinutesPerWeek;
        this.maximumMinutesPerMonth = maximumMinutesPerMonth;
        this.maximumMinutesPerYear = maximumMinutesPerYear;
    }

    @AssertTrue
    @JsonIgnore
    public boolean isValid() {
        if (maximumMinutesPerDay != null && maximumMinutesPerDay <= 0) {
            return false;
        }
        if (maximumMinutesPerWeek != null && maximumMinutesPerWeek <= 0) {
            return false;
        }
        if (maximumMinutesPerMonth != null && maximumMinutesPerMonth <= 0) {
            return false;
        }
        if (maximumMinutesPerYear != null && maximumMinutesPerYear <= 0) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return name;
    }

    // ************************************************************************
    // Simple getters and setters
    // ************************************************************************

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getMaximumMinutesPerDay() {
        return maximumMinutesPerDay;
    }

    public void setMaximumMinutesPerDay(Integer maximumMinutesPerDay) {
        this.maximumMinutesPerDay = maximumMinutesPerDay;
    }

    public Integer getMaximumMinutesPerWeek() {
        return maximumMinutesPerWeek;
    }

    public void setMaximumMinutesPerWeek(Integer maximumMinutesPerWeek) {
        this.maximumMinutesPerWeek = maximumMinutesPerWeek;
    }

    public Integer getMaximumMinutesPerMonth() {
        return maximumMinutesPerMonth;
    }

    public void setMaximumMinutesPerMonth(Integer maximumMinutesPerMonth) {
        this.maximumMinutesPerMonth = maximumMinutesPerMonth;
    }

    public Integer getMaximumMinutesPerYear() {
        return maximumMinutesPerYear;
    }

    public void setMaximumMinutesPerYear(Integer maximumMinutesPerYear) {
        this.maximumMinutesPerYear = maximumMinutesPerYear;
    }
}
