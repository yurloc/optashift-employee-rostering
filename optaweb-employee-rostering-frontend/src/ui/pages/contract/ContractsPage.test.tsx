/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import OptionalInput from 'ui/components/OptionalInput';
import { Sorter } from 'types';
import { Contract } from 'domain/Contract';
import { useTranslation } from 'react-i18next';
import { getRouterProps } from 'util/BookmarkableTestUtils';
import { ContractsPage, Props } from './ContractsPage';

describe('Contracts page', () => {
  it('should render correctly with no contracts', () => {
    const contractsPage = shallow(<ContractsPage {...noContracts} />);
    expect(toJson(contractsPage)).toMatchSnapshot();
  });

  it('should render correctly with a few contracts', () => {
    const contractsPage = shallow(<ContractsPage {...twoContracts} />);
    expect(toJson(contractsPage)).toMatchSnapshot();
  });

  it('should render the viewer correctly', () => {
    const contractsPage = new ContractsPage(twoContracts);
    const contract = twoContracts.tableData[1];
    const viewer = shallow(contractsPage.renderViewer(contract));
    expect(toJson(viewer)).toMatchSnapshot();
  });

  it('should render the editor correctly', () => {
    const contractsPage = new ContractsPage(twoContracts);
    const contract = twoContracts.tableData[1];
    const editor = shallow(contractsPage.renderEditor(contract));
    expect(toJson(editor)).toMatchSnapshot();
  });

  it('should update properties on change', () => {
    const spotsPage = new ContractsPage(twoContracts);
    const setProperty = jest.fn();
    const editor = spotsPage.editDataRow(spotsPage.getInitialStateForNewRow(), setProperty);
    const nameCol = shallow(editor[0]);
    nameCol.simulate('change', { currentTarget: { value: 'Test' } });
    expect(setProperty).toBeCalled();
    expect(setProperty).toBeCalledWith('name', 'Test');

    setProperty.mockClear();
    const maxMinutesPerDayCol = mount(editor[1]).find(OptionalInput);
    expect(maxMinutesPerDayCol.props().valueToString(10)).toEqual('10');
    expect(maxMinutesPerDayCol.props().valueMapper('10')).toEqual(10);
    expect(maxMinutesPerDayCol.props().isValid('10')).toEqual(true);
    expect(maxMinutesPerDayCol.props().isValid('ab10')).toEqual(false);
    expect(maxMinutesPerDayCol.props().isValid('-10')).toEqual(false);
    maxMinutesPerDayCol.find(OptionalInput).props().onChange(10);
    expect(setProperty).toBeCalled();
    expect(setProperty).toBeCalledWith('maximumMinutesPerDay', 10);

    setProperty.mockClear();
    const maxMinutesPerWeekCol = mount(editor[2]).find(OptionalInput);
    expect(maxMinutesPerWeekCol.props().valueToString(10)).toEqual('10');
    expect(maxMinutesPerWeekCol.props().valueMapper('10')).toEqual(10);
    expect(maxMinutesPerWeekCol.props().isValid('10')).toEqual(true);
    expect(maxMinutesPerWeekCol.props().isValid('ab10')).toEqual(false);
    expect(maxMinutesPerWeekCol.props().isValid('-10')).toEqual(false);
    maxMinutesPerWeekCol.props().onChange(10);
    expect(setProperty).toBeCalled();
    expect(setProperty).toBeCalledWith('maximumMinutesPerWeek', 10);

    setProperty.mockClear();
    const maxMinutesPerMonthCol = mount(editor[3]).find(OptionalInput);
    expect(maxMinutesPerMonthCol.props().valueToString(10)).toEqual('10');
    expect(maxMinutesPerMonthCol.props().valueMapper('10')).toEqual(10);
    expect(maxMinutesPerMonthCol.props().isValid('10')).toEqual(true);
    expect(maxMinutesPerMonthCol.props().isValid('ab10')).toEqual(false);
    expect(maxMinutesPerMonthCol.props().isValid('-10')).toEqual(false);
    maxMinutesPerMonthCol.props().onChange(10);
    expect(setProperty).toBeCalled();
    expect(setProperty).toBeCalledWith('maximumMinutesPerMonth', 10);

    setProperty.mockClear();
    const maxMinutesPerYearCol = mount(editor[4]).find(OptionalInput);
    expect(maxMinutesPerYearCol.props().valueToString(10)).toEqual('10');
    expect(maxMinutesPerYearCol.props().valueMapper('10')).toEqual(10);
    expect(maxMinutesPerYearCol.props().isValid('10')).toEqual(true);
    expect(maxMinutesPerYearCol.props().isValid('ab10')).toEqual(false);
    expect(maxMinutesPerYearCol.props().isValid('-10')).toEqual(false);
    maxMinutesPerYearCol.props().onChange(10);
    expect(setProperty).toBeCalled();
    expect(setProperty).toBeCalledWith('maximumMinutesPerYear', 10);
  });

  it('should call addContract on addData', () => {
    const contractsPage = new ContractsPage(twoContracts);
    const contract = twoContracts.tableData[1];
    contractsPage.addData(contract);
    expect(twoContracts.addContract).toBeCalled();
    expect(twoContracts.addContract).toBeCalledWith(contract);
  });

  it('should call updateContract on updateData', () => {
    const contractsPage = new ContractsPage(twoContracts);
    const contract = twoContracts.tableData[1];
    contractsPage.updateData(contract);
    expect(twoContracts.updateContract).toBeCalled();
    expect(twoContracts.updateContract).toBeCalledWith(contract);
  });

  it('should call removeContract on removeData', () => {
    const contractsPage = new ContractsPage(twoContracts);
    const contract = twoContracts.tableData[1];
    contractsPage.removeData(contract);
    expect(twoContracts.removeContract).toBeCalled();
    expect(twoContracts.removeContract).toBeCalledWith(contract);
  });

  it('should return a filter that match by name', () => {
    const contractPage = new ContractsPage(twoContracts);
    const filter = contractPage.getFilter();

    expect(twoContracts.tableData.filter(filter('1'))).toEqual([twoContracts.tableData[0]]);
    expect(twoContracts.tableData.filter(filter('2'))).toEqual([twoContracts.tableData[1]]);
  });

  it('should return a sorter that sort by name', () => {
    const contractPage = new ContractsPage(twoContracts);
    const sorter = contractPage.getSorters()[0] as Sorter<Contract>;
    const list = [twoContracts.tableData[1], twoContracts.tableData[0]];
    expect(list.sort(sorter)).toEqual(twoContracts.tableData);
  });

  it('should treat incompleted data as incomplete', () => {
    const contractsPage = new ContractsPage(twoContracts);

    const noName = { ...twoContracts.tableData[1], name: undefined };
    const result1 = contractsPage.isDataComplete(noName);
    expect(result1).toEqual(false);

    const noMaxHoursPerDay = { ...twoContracts.tableData[1], maximumMinutesPerDay: undefined };
    const result2 = contractsPage.isDataComplete(noMaxHoursPerDay);
    expect(result2).toEqual(false);

    const noMaxHoursPerWeek = { ...twoContracts.tableData[1], maximumMinutesPerWeek: undefined };
    const result3 = contractsPage.isDataComplete(noMaxHoursPerWeek);
    expect(result3).toEqual(false);

    const noMaxHoursPerMonth = { ...twoContracts.tableData[1], maximumMinutesPerMonth: undefined };
    const result4 = contractsPage.isDataComplete(noMaxHoursPerMonth);
    expect(result4).toEqual(false);

    const noMaxHoursPerYear = { ...twoContracts.tableData[1], maximumMinutesPerYear: undefined };
    const result5 = contractsPage.isDataComplete(noMaxHoursPerYear);
    expect(result5).toEqual(false);

    const completed = { ...twoContracts.tableData[1] };
    const result6 = contractsPage.isDataComplete(completed);
    expect(result6).toEqual(true);
  });

  it('should treat empty name as invalid', () => {
    const contractsPage = new ContractsPage(twoContracts);
    const contract = { ...twoContracts.tableData[1], name: '' };
    const result = contractsPage.isValid(contract);
    expect(result).toEqual(false);
  });

  it('should treat non-empty name as valid', () => {
    const contractsPage = new ContractsPage(twoContracts);
    const contract = { ...twoContracts.tableData[1], name: 'Contract' };
    const result = contractsPage.isValid(contract);
    expect(result).toEqual(true);
  });
});

const noContracts: Props = {
  ...useTranslation('ContractsPage'),
  tReady: true,
  tenantId: 0,
  title: 'Contracts',
  columnTitles: ['Name', 'Max Hours Per Day', 'Max Hours Per Week', 'Max Hours Per Month', 'Max Hours Per Year'],
  tableData: [],
  addContract: jest.fn(),
  updateContract: jest.fn(),
  removeContract: jest.fn(),
  ...getRouterProps('/contacts', {}),
};

const twoContracts: Props = {
  ...useTranslation('ContractsPage'),
  tReady: true,
  tenantId: 0,
  title: 'Contracts',
  columnTitles: ['Name', 'Max Hours Per Day', 'Max Hours Per Week', 'Max Hours Per Month', 'Max Hours Per Year'],
  tableData: [{
    id: 0,
    version: 0,
    tenantId: 0,
    name: 'Contract 1',
    maximumMinutesPerDay: null,
    maximumMinutesPerWeek: null,
    maximumMinutesPerMonth: null,
    maximumMinutesPerYear: null,
  },
  {
    id: 1,
    version: 0,
    tenantId: 0,
    name: 'Contract 2',
    maximumMinutesPerDay: 1,
    maximumMinutesPerWeek: 20,
    maximumMinutesPerMonth: 10,
    maximumMinutesPerYear: 120,
  }],
  addContract: jest.fn(),
  updateContract: jest.fn(),
  removeContract: jest.fn(),
  ...getRouterProps('/contacts', {}),
};
