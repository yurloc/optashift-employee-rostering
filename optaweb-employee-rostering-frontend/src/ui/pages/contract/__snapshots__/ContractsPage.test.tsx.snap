// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contracts page should render correctly with a few contracts 1`] = `
<Fragment>
  <Level
    gutter="sm"
    style={
      Object {
        "backgroundColor": "var(--pf-global--BackgroundColor--200)",
        "padding": "5px 5px 5px 5px",
      }
    }
  >
    <LevelItem>
      <FilterComponent
        filterText=""
        onChange={[Function]}
      />
    </LevelItem>
    <LevelItem
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <Button
        isDisabled={false}
        onClick={[Function]}
      >
        Trans(i18nKey=add)
      </Button>
      <Pagination
        itemCount={2}
        onPerPageSelect={[Function]}
        onSetPage={[Function]}
        page={1}
        perPage={10}
        widgetId="pagination-options-menu-top"
      />
    </LevelItem>
  </Level>
  <Table
    borders={true}
    caption="Contracts"
    cells={
      Array [
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "Name",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "Max Hours Per Day",
          "transforms": undefined,
        },
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "Max Hours Per Week",
          "transforms": undefined,
        },
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "Max Hours Per Month",
          "transforms": undefined,
        },
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "Max Hours Per Year",
          "transforms": undefined,
        },
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "",
          "transforms": undefined,
        },
      ]
    }
    className=""
    contentId="expanded-content"
    dropdownDirection="down"
    dropdownPosition="right"
    expandId="expandable-toggle"
    gridBreakPoint="grid-md"
    onSort={[Function]}
    role="grid"
    rowLabeledBy="simple-node"
    rows={
      Array [
        Object {
          "cells": Array [
            Object {
              "title": <EditableComponent
                editor={
                  <TextInput
                    aria-label="Name"
                    className=""
                    defaultValue="Contract 1"
                    isDisabled={false}
                    isReadOnly={false}
                    isRequired={false}
                    isValid={true}
                    name="name"
                    onChange={[Function]}
                    type="text"
                  />
                }
                isEditing={false}
                viewer={
                  <Text>
                    Contract 1
                  </Text>
                }
              />,
            },
            Object {
              "title": <EditableComponent
                editor={
                  <OptionalInput
                    defaultValue={null}
                    isValid={[Function]}
                    label="Trans(i18nKey=maxMinutesPerDay)"
                    onChange={[Function]}
                    valueMapper={[Function]}
                    valueToString={[Function]}
                  />
                }
                isEditing={false}
                viewer={<Text />}
              />,
            },
            Object {
              "title": <EditableComponent
                editor={
                  <OptionalInput
                    defaultValue={null}
                    isValid={[Function]}
                    label="Trans(i18nKey=maxMinutesPerWeek)"
                    onChange={[Function]}
                    valueMapper={[Function]}
                    valueToString={[Function]}
                  />
                }
                isEditing={false}
                viewer={<Text />}
              />,
            },
            Object {
              "title": <EditableComponent
                editor={
                  <OptionalInput
                    defaultValue={null}
                    isValid={[Function]}
                    label="Trans(i18nKey=maxMinutesPerMonth)"
                    onChange={[Function]}
                    valueMapper={[Function]}
                    valueToString={[Function]}
                  />
                }
                isEditing={false}
                viewer={<Text />}
              />,
            },
            Object {
              "title": <EditableComponent
                editor={
                  <OptionalInput
                    defaultValue={null}
                    isValid={[Function]}
                    label="Trans(i18nKey=maxMinutesPerYear)"
                    onChange={[Function]}
                    valueMapper={[Function]}
                    valueToString={[Function]}
                  />
                }
                isEditing={false}
                viewer={<Text />}
              />,
            },
            Object {
              "title": <EditableComponent
                editor={
                  <span
                    style={
                      Object {
                        "display": "grid",
                        "gridColumnGap": "5px",
                        "gridTemplateColumns": "1fr auto auto",
                      }
                    }
                  >
                    <span />
                    <Button
                      aria-label="Save"
                      onClick={[Function]}
                      variant="link"
                    >
                      <SaveIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      />
                    </Button>
                    <Button
                      aria-label="Cancel"
                      onClick={[Function]}
                      variant="link"
                    >
                      <CloseIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      />
                    </Button>
                  </span>
                }
                isEditing={false}
                viewer={
                  <span
                    style={
                      Object {
                        "display": "grid",
                        "gridColumnGap": "5px",
                        "gridTemplateColumns": "1fr auto auto",
                      }
                    }
                  >
                    <span />
                    <Button
                      aria-label="Edit"
                      onClick={[Function]}
                      variant="link"
                    >
                      <EditIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      />
                    </Button>
                    <Button
                      aria-label="Delete"
                      onClick={[Function]}
                      variant="link"
                    >
                      <TrashIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      />
                    </Button>
                  </span>
                }
              />,
            },
          ],
        },
        Object {
          "cells": Array [
            Object {
              "title": <EditableComponent
                editor={
                  <TextInput
                    aria-label="Name"
                    className=""
                    defaultValue="Contract 2"
                    isDisabled={false}
                    isReadOnly={false}
                    isRequired={false}
                    isValid={true}
                    name="name"
                    onChange={[Function]}
                    type="text"
                  />
                }
                isEditing={false}
                viewer={
                  <Text>
                    Contract 2
                  </Text>
                }
              />,
            },
            Object {
              "title": <EditableComponent
                editor={
                  <OptionalInput
                    defaultValue={1}
                    isValid={[Function]}
                    label="Trans(i18nKey=maxMinutesPerDay)"
                    onChange={[Function]}
                    valueMapper={[Function]}
                    valueToString={[Function]}
                  />
                }
                isEditing={false}
                viewer={
                  <Text>
                    1
                  </Text>
                }
              />,
            },
            Object {
              "title": <EditableComponent
                editor={
                  <OptionalInput
                    defaultValue={20}
                    isValid={[Function]}
                    label="Trans(i18nKey=maxMinutesPerWeek)"
                    onChange={[Function]}
                    valueMapper={[Function]}
                    valueToString={[Function]}
                  />
                }
                isEditing={false}
                viewer={
                  <Text>
                    20
                  </Text>
                }
              />,
            },
            Object {
              "title": <EditableComponent
                editor={
                  <OptionalInput
                    defaultValue={10}
                    isValid={[Function]}
                    label="Trans(i18nKey=maxMinutesPerMonth)"
                    onChange={[Function]}
                    valueMapper={[Function]}
                    valueToString={[Function]}
                  />
                }
                isEditing={false}
                viewer={
                  <Text>
                    10
                  </Text>
                }
              />,
            },
            Object {
              "title": <EditableComponent
                editor={
                  <OptionalInput
                    defaultValue={120}
                    isValid={[Function]}
                    label="Trans(i18nKey=maxMinutesPerYear)"
                    onChange={[Function]}
                    valueMapper={[Function]}
                    valueToString={[Function]}
                  />
                }
                isEditing={false}
                viewer={
                  <Text>
                    120
                  </Text>
                }
              />,
            },
            Object {
              "title": <EditableComponent
                editor={
                  <span
                    style={
                      Object {
                        "display": "grid",
                        "gridColumnGap": "5px",
                        "gridTemplateColumns": "1fr auto auto",
                      }
                    }
                  >
                    <span />
                    <Button
                      aria-label="Save"
                      onClick={[Function]}
                      variant="link"
                    >
                      <SaveIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      />
                    </Button>
                    <Button
                      aria-label="Cancel"
                      onClick={[Function]}
                      variant="link"
                    >
                      <CloseIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      />
                    </Button>
                  </span>
                }
                isEditing={false}
                viewer={
                  <span
                    style={
                      Object {
                        "display": "grid",
                        "gridColumnGap": "5px",
                        "gridTemplateColumns": "1fr auto auto",
                      }
                    }
                  >
                    <span />
                    <Button
                      aria-label="Edit"
                      onClick={[Function]}
                      variant="link"
                    >
                      <EditIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      />
                    </Button>
                    <Button
                      aria-label="Delete"
                      onClick={[Function]}
                      variant="link"
                    >
                      <TrashIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      />
                    </Button>
                  </span>
                }
              />,
            },
          ],
        },
      ]
    }
    sortBy={Object {}}
    variant={null}
  >
    <TableHeader />
    <TableBody />
  </Table>
</Fragment>
`;

exports[`Contracts page should render correctly with no contracts 1`] = `
<Fragment>
  <Level
    gutter="sm"
    style={
      Object {
        "backgroundColor": "var(--pf-global--BackgroundColor--200)",
        "padding": "5px 5px 5px 5px",
      }
    }
  >
    <LevelItem>
      <FilterComponent
        filterText=""
        onChange={[Function]}
      />
    </LevelItem>
    <LevelItem
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <Button
        isDisabled={false}
        onClick={[Function]}
      >
        Trans(i18nKey=add)
      </Button>
      <Pagination
        itemCount={0}
        onPerPageSelect={[Function]}
        onSetPage={[Function]}
        page={1}
        perPage={10}
        widgetId="pagination-options-menu-top"
      />
    </LevelItem>
  </Level>
  <Table
    borders={true}
    caption="Contracts"
    cells={
      Array [
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "Name",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "Max Hours Per Day",
          "transforms": undefined,
        },
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "Max Hours Per Week",
          "transforms": undefined,
        },
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "Max Hours Per Month",
          "transforms": undefined,
        },
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "Max Hours Per Year",
          "transforms": undefined,
        },
        Object {
          "cellTransforms": Array [
            [Function],
          ],
          "props": Object {},
          "title": "",
          "transforms": undefined,
        },
      ]
    }
    className=""
    contentId="expanded-content"
    dropdownDirection="down"
    dropdownPosition="right"
    expandId="expandable-toggle"
    gridBreakPoint="grid-md"
    onSort={[Function]}
    role="grid"
    rowLabeledBy="simple-node"
    rows={Array []}
    sortBy={Object {}}
    variant={null}
  >
    <TableHeader />
    <TableBody />
  </Table>
</Fragment>
`;

exports[`Contracts page should render the editor correctly 1`] = `
<tr>
  <td
    key="0"
  >
    <TextInput
      aria-label="Name"
      className=""
      defaultValue="Contract 2"
      isDisabled={false}
      isReadOnly={false}
      isRequired={false}
      isValid={true}
      key="0"
      name="name"
      onChange={[Function]}
      type="text"
    />
  </td>
  <td
    key="1"
  >
    <OptionalInput
      defaultValue={1}
      isValid={[Function]}
      key="1"
      label="Trans(i18nKey=maxMinutesPerDay)"
      onChange={[Function]}
      valueMapper={[Function]}
      valueToString={[Function]}
    />
  </td>
  <td
    key="2"
  >
    <OptionalInput
      defaultValue={20}
      isValid={[Function]}
      key="2"
      label="Trans(i18nKey=maxMinutesPerWeek)"
      onChange={[Function]}
      valueMapper={[Function]}
      valueToString={[Function]}
    />
  </td>
  <td
    key="3"
  >
    <OptionalInput
      defaultValue={10}
      isValid={[Function]}
      key="3"
      label="Trans(i18nKey=maxMinutesPerMonth)"
      onChange={[Function]}
      valueMapper={[Function]}
      valueToString={[Function]}
    />
  </td>
  <td
    key="4"
  >
    <OptionalInput
      defaultValue={120}
      isValid={[Function]}
      key="4"
      label="Trans(i18nKey=maxMinutesPerYear)"
      onChange={[Function]}
      valueMapper={[Function]}
      valueToString={[Function]}
    />
  </td>
</tr>
`;

exports[`Contracts page should render the viewer correctly 1`] = `
<tr>
  <td
    key="0"
  >
    <Text
      key="0"
    >
      Contract 2
    </Text>
  </td>
  <td
    key="1"
  >
    <Text
      key="1"
    >
      1
    </Text>
  </td>
  <td
    key="2"
  >
    <Text
      key="2"
    >
      20
    </Text>
  </td>
  <td
    key="3"
  >
    <Text
      key="3"
    >
      10
    </Text>
  </td>
  <td
    key="4"
  >
    <Text
      key="4"
    >
      120
    </Text>
  </td>
</tr>
`;
